src/components/TextArea/TextArea.jsx

```diff
@@ -8,55 +8,74 @@ import style from './style.scss';
 
 class TextArea extends Component {
   render() {
+    const {
+      data,
+      disabled,
+      error,
+      maxLength,
+      onBlur,
+      onChange,
+      onFocus,
+      placeholder,
+      rows,
+      size,
+      value,
+    } = this.props;
+
     const wrapperStyle = cx(style.wrapper, {
-      [style.wrapperDisabled]: this.props.disabled,
-      [style.wrapperError]: this.props.error,
-      [style.wrapperLarge]: toUpper(this.props.size) === 'LARGE',
+      [style.wrapperDisabled]: disabled,
+      [style.wrapperError]: error,
+      [style.wrapperLarge]: toUpper(size) === 'LARGE',
     });
 
     const textareaStyle = cx(style.textarea, {
-      [style.disabled]: this.props.disabled,
-      [style.large]: toUpper(this.props.size) === 'LARGE',
+      [style.disabled]: disabled,
+      [style.large]: toUpper(size) === 'LARGE',
     });
 
     return (
-      <div className={wrapperStyle} {...getDataAttrs(this.props.data)}>
+      <div className={wrapperStyle} {...getDataAttrs(data)}>
         <textarea
-          ref={el => el && new Autogrow(el)}
+          ref={el => {
+            if (el && !el.props) {
```

Comment:
why would el have props? its a dom element?

Reply:
it's either dom element (which we need) or an instance; i'm not sure why React does this but it seems to interchange in my console